File: tests/arch/fixtures/real_world/csharp/src/Result/Success.cs
using Example.Result.IResult;

namespace Example.Result;

/// <summary>Represents a successful result wrapping a value.</summary>
public sealed class Success<T> : IResult<T>
{
    public Success(T value) => Value = value;

    public bool IsSuccess => true;
    public T Value { get; }
    public string? Error => null;

    public override string ToString() => $"Success({Value})";
}

File: tests/arch/fixtures/real_world/php/src/Result/Success.php
<?php

declare(strict_types=1);

namespace Example\Result;

use Example\Result\IResult;

/**
 * Represents a successful result wrapping a value.
 */
final class Success implements IResult
{
    /** @param mixed $value */
    public function __construct(private readonly mixed $value) {}

    public function isSuccess(): bool
    {
        return true;
    }

    /** @return mixed */
    public function getValue(): mixed
    {
        return $this->value;
    }

    public function getError(): ?string
    {
        return null;
    }

    public function __toString(): string
    {
        return sprintf('Success(%s)', $this->value);
    }
}
